File: spec/python/test_if_struct.py
import unittest

from if_struct import IfStruct

class TestIfStruct(unittest.TestCase):
    def test_if_struct(self):
        r = IfStruct.from_file("src/if_struct.bin")

        self.assertEqual(r.op1.opcode, 0x53)
        self.assertEqual(r.op1.arg_str.str, "foo")

        self.assertEqual(r.op2.opcode, 0x54)
        self.assertEqual(r.op2.arg_tuple.num1, 0x42)
        self.assertEqual(r.op2.arg_tuple.num2, 0x43)

        self.assertEqual(r.op3.opcode, 0x53)
        self.assertEqual(r.op3.arg_str.str, "bar")

File: spec/javascript/test_if_struct.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('IfStruct', 'src/if_struct.bin', function(r) {
    assert.equal(r.op1.opcode, 0x53);
    assert.equal(r.op1.argStr.str, "foo");

    assert.equal(r.op2.opcode, 0x54);
    assert.equal(r.op2.argTuple.num1, 0x42);
    assert.equal(r.op2.argTuple.num2, 0x43);

    assert.equal(r.op3.opcode, 0x53);
    assert.equal(r.op3.argStr.str, "bar");
})
